packages/shared/src/contexts/PaymentContext.tsx

```diff
@@ -46,12 +55,46 @@ export const PaymentContextProvider = ({
         (process.env.NEXT_PUBLIC_PADDLE_ENVIRONMENT as Environments) ||
         'production',
       token: process.env.NEXT_PUBLIC_PADDLE_TOKEN,
+      eventCallback: (event: PaddleEventData) => {
+        switch (event?.name) {
+          case CheckoutEventNames.CHECKOUT_PAYMENT_SELECTED:
+            logSubscriptionEvent({
+              event_name: LogEvent.SelectCheckoutPayment,
+              target_id: event?.data?.payment.method_details.type,
+            });
+            break;
+          case CheckoutEventNames.CHECKOUT_COMPLETED:
+            logSubscriptionEvent({
+              event_name: LogEvent.CompleteCheckout,
+              extra: {
+                cycle: event?.data.items?.[0]?.billing_cycle.interval,
+                cost: event?.data.totals.total,
+                currency: event?.data.currency_code,
+                payment: event?.data.payment.method_details.type,
+              },
+            });
+            break;
+          // This doesn't exist in the original code
+          case 'checkout.warning' as CheckoutEventNames:
+            logSubscriptionEvent({
+              event_name: LogEvent.WarningCheckout,
+            });
+            break;
+          case CheckoutEventNames.CHECKOUT_ERROR:
+            logSubscriptionEvent({
+              event_name: LogEvent.ErrorCheckout,
+            });
+            break;
```

Comment:
Waiting for some clarification on these, as they don't contain anything from the taxonomy